fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -179,6 +183,47 @@ private IndexingByIndex getIndexerByIndex() {
         if (ex == null) {
             return AsyncUtil.DONE;
         }
+        if (ex.getCause() instanceof IndexingBase.PartlyBuiltException) {
+            IndexBuildProto.IndexBuildIndexingStamp conflictingIndexingTypeStamp = ((IndexingBase.PartlyBuiltException)ex.getCause()).savedStamp;
+            // Here: an indexing type stamp already exists, with a different type. Few precondition cases are handled
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info(KeyValueLogMessage.build("conflicting indexing type stamp",
+                        LogMessageKeys.INDEX_NAME, common.getIndex().getName(),
+                        LogMessageKeys.INDEX_VERSION, common.getIndex().getLastModifiedVersion(),
+                        LogMessageKeys.INDEXER_ID, common.getUuid(),
+                        LogMessageKeys.INDEX_STATE_PRECONDITION, common.getIndexStatePrecondition(),
+                        LogMessageKeys.ACTUAL_TYPE, conflictingIndexingTypeStamp
+                ).toString());
+            }
+            if (common.getIndexStatePrecondition() == IndexStatePrecondition.BUILD_IF_DISABLED_CONTINUE_BUILD_IF_WRITE_ONLY) {
+                indexer = null;
+                // now, match the policy to the previous run
+                IndexBuildProto.IndexBuildIndexingStamp.Method method = conflictingIndexingTypeStamp.getMethod();
+                if ( method == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_RECORDS) {
+                    fallbackToRecordsScan = true;
+                    return fallbackFunction.get();
+                }
+                if (method == IndexBuildProto.IndexBuildIndexingStamp.Method.BY_INDEX) {
+                    Object sourceIndexSubspaceKey = decodeSubspaceKey(conflictingIndexingTypeStamp.getSourceIndexSubspaceKey());
+                    indexFromIndexPolicy = IndexFromIndexPolicy.newBuilder()
+                            .setSourceIndexSubspaceKey(sourceIndexSubspaceKey)
```

Comment:
From a previous comment: https://github.com/FoundationDB/fdb-record-layer/pull/1217#discussion_r608277922

> @alecgrieser: Also, is there a test that exercises this code path (i.e., resumes a build with a different index from the original one provided)?
> @jjezra: `testIndexFromIndexPersistentContinuePreviousByIndex` should be covering this code.

I think that that test handles the case of "first index build specifies a source index; second index build specifies by records". I think that's a good test case to have, but I was also interested in the cases of "first index build specifies a source index; second index build specifies a different source index" or even "first index build specifies a source index; second index build specifies a newer source index (i.e., the `lastModifiedVersion` has changed)".

Reply:
Added `testIndexFromIndexSrcVersionModifiedWithFallback`.